Objects/unicodeobject.c

```diff
@@ -14022,7 +14022,7 @@ _PyUnicodeWriter_WriteASCIIString(_PyUnicodeWriter *writer,
                                   const char *ascii, Py_ssize_t len)
 {
     if (len == -1)
-        len = strlen(ascii);
+        len = strnlen(ascii, len);
```

Comment:
I don't think that passing -1 to `strnlen()` is correct.

Reply:
Good catch! Will fix.